async-nats/src/subject.rs

```diff
@@ -110,3 +110,25 @@ impl fmt::Display for Subject {
         write!(f, "{}", self.as_str())
     }
 }
+
+pub trait AsSubject {
+    fn as_subject(&self) -> Subject;
+}
+
+impl AsSubject for Subject {
+    fn as_subject(&self) -> Subject {
+        self.to_owned()
+    }
+}
+
+impl AsSubject for &'static str {
+    fn as_subject(&self) -> Subject {
+        Subject::from_static(self)
+    }
+}
+
+impl AsSubject for String {
```

Comment:
Should we also have an implementation for &[u8] and Vec<u8> so that if someone's already got a blob for their subject they don't need to convert it?

Reply:
This is a good idea.
Will make a follow up.